### DISCUSSION TITLE
How to delete all another polygons?

### DISCUSSION THREAD

**Discussion by valeriu-dev** [geoman-io/leaflet-geoman#862] [Questions & Help]
Hello, need to remove all another polygons when user add a new polygon on map.
Or limit count of polygons to 1.
Can someone tell me how to do this?
```
this.map.on('pm:create', e => {
     // I listen the create event
});
```

I tried: this.map.pm.Draw.Polygon._removeLastVertex() but get an error:
![Screenshot_2](https://user-images.githubusercontent.com/14403446/57050241-4f188280-6c84-11e9-8d67-440997c00f0d.png)

**Comment by codeofsumit** [comment]
You can do this with regular leaflet. Something like this:

```js
this.map.on('pm:create', e => {
  this.map.eachLayer(layer, () => {
    if(!e.target !== layer) {
      layer.remove();
    }
  })
});
```

**Comment by valeriu-dev** [comment]
@codeofsumit Thank You!

My working code is:
```
this.map.eachLayer(layer => {
  if (typeof layer._latlngs !== 'undefined' && layer._latlngs.length > 0 && e.layer._leaflet_id !== layer._leaflet_id) {
	layer.remove()
  }
})
```
